auth/src/main/java/com/firebase/ui/auth/viewmodel/credentialmanager/CredentialManagerHandler.kt

```diff
@@ -0,0 +1,87 @@
+package com.firebase.ui.auth.viewmodel.credentialmanager
+
+import android.app.Application
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.credentials.CreatePasswordRequest
+import androidx.credentials.CredentialManager
+import androidx.credentials.exceptions.CreateCredentialException
+import com.firebase.ui.auth.ErrorCodes
+import com.firebase.ui.auth.FirebaseUiException
+import com.firebase.ui.auth.IdpResponse
+import com.firebase.ui.auth.data.model.Resource
+import com.firebase.ui.auth.viewmodel.AuthViewModelBase
+import com.google.firebase.auth.FirebaseUser
+import kotlinx.coroutines.launch
+
+class CredentialManagerHandler(application: Application) :
+    AuthViewModelBase<IdpResponse>(application) {
+
+    private val credentialManager = CredentialManager.create(application)
+    private var response: IdpResponse? = null
+
+    fun setResponse(newResponse: IdpResponse) {
+        response = newResponse
+    }
+
+    /**
+     * Saves credentials via Credential Manager if enabled in [getArguments().enableCredentials].
+     * Uses a password-based credential for demonstration; adapt to passkeys or other flows as needed.
+     */
+    fun saveCredentials(
+        activity: androidx.activity.ComponentActivity,
+        firebaseUser: FirebaseUser?,
+        password: String?
+    ) {
+        if (!arguments.enableCredentials) {
+            setResult(Resource.forSuccess(response!!))
+            return
+        }
+        setResult(Resource.forLoading())
+
+        if (firebaseUser == null || firebaseUser.email.isNullOrEmpty() || password.isNullOrEmpty()) {
+            setResult(
+                Resource.forFailure(
+                    FirebaseUiException(
+                        ErrorCodes.UNKNOWN_ERROR,
+                        "Invalid FirebaseUser or missing password."
+                    )
+                )
+            )
+            return
+        }
+
+        // Example: Password credential with the user's email as the identifier
+        val request = CreatePasswordRequest(
+            id = firebaseUser.email!!,
+            password = password
+        )
+
+        viewModelScope.launch {
+            try {
+                credentialManager.createCredential(activity, request)
```

Comment:
Looks like [`createCredential()`](https://developer.android.com/reference/android/credentials/CredentialManager#createCredential(android.content.Context,%20android.credentials.CreateCredentialRequest,%20android.os.CancellationSignal,%20java.util.concurrent.Executor,%20android.os.OutcomeReceiver%3Candroid.credentials.CreateCredentialResponse,android.credentials.CreateCredentialException%3E)) takes `Context` as argument. Maybe we can pass that to `saveCredentials()` instead of `activity: androidx.activity.ComponentActivity` ? 

Also, do we care about the result of `createCredential()` ? it seems like we're ignoring it right now.

Reply:
From what I understood, we don't need to get the result - if `createCredential()` succeeds, we've safely saved the user's password. But that brings me to another question: (out of curiosity) what's _in_ the `result` of this call?